diplomacy/engine/game.py

```diff
@@ -893,7 +893,7 @@ def new_power_message(self, recipient, body):
         assert self.is_player_game()
         if not self.has_power(recipient):
             raise exceptions.MapPowerException(recipient)
-        return Message(phase=self.current_short_phase, sender=self.role, recipient=recipient, message=body)
+        return Message(phase=self.current_short_phase, sender=self.role, recipient=recipient, message=body, thread_idx=thread_idx)
```

Comment:
@bkj could you add the `thread_idx` param to the docstring? Otherwise, the only exploration for it would be in this PR.